lib/animas/animasSimulator.js

```diff
@@ -85,9 +90,53 @@ exports.make = function(config){
     events.push(e);
   }
 
+  function checkIfBasalSuspendedByAlarm(alarm,basal) {
+    if ((basal !== null) && (alarm != null) &&
+        (basal.rate === 0) && (basal.deliveryType !== 'suspend')) {
+
+      // Animas does not generate suspend/resume events for alarms,
+      // so we check if an alarm occurred recently or at the same time.
+      // If so, we mark the basal as suspended and generate a new suspend event.
+
+      var duration = Date.parse(basal.time) - Date.parse(alarm.time);
+      console.log(alarm, basal);
+      var twoHours = (120 * sundial.MIN_TO_MSEC);
+      if(duration < twoHours) {
+
+        var type = alarm.alarmType;
+        if(type === 'occlusion' || type === 'auto_off' || type === 'no_insulin' || type === 'no_power') {
+
+          //this basal is suspended
+          basal.deliveryType = 'suspend';
+          annotate.annotateEvent(basal, 'animas/basal/marked-suspended-from-alarm');
+
+          var status = {
+            time: basal.time,
+            deviceTime: basal.deviceTime,
```

Comment:
I don't know, when the `deviceTime` thing tripped me up, it did _not_ match the times in my pump history that I was scrolling through. (I had a bolus showing a `deviceTime` of between 4-5 a.m. when I was definitely not awake.) Unless you're saying that the issue with `time` was having some kind of a cascading effect? Let me know if/when/on what branch you want me to test again to see...